ols/src/cache/postgres_cache.py

```diff
@@ -0,0 +1,142 @@
+"""Cache that uses Postgres to store cached values."""
+
+import pickle
+from typing import Optional
+
+from langchain_core.messages.base import BaseMessage
+
+from ols.src.cache.cache import Cache
+
+
+class PostgresCache(Cache):
+    """Cache that uses Postgres to store cached values."""
+
+    CREATE_CACHE_TABLE = """
+        CREATE TABLE IF NOT EXISTS cache (
+            user_id         text NOT NULL,
+            conversation_id text NOT NULL,
+            key             text UNIQUE NOT NULL,
+            value           bytea,
+            updated_at      timestamp,
+            PRIMARY KEY(user_id, conversation_id)
+        );
+        """
+
+    CREATE_INDEX = """
+        CREATE INDEX IF NOT EXISTS timestamps
+            ON cache (updated_at)
+        """
+
+    SELECT_STATEMENT = """
+        SELECT value
+          FROM cache
+         WHERE user_id=%s and conversation_id=%s limit 1
```

Comment:
Upper `LIMIT`?

Reply:
WDYM please?

Reply:
` WHERE user_id=%s and conversation_id=%s LIMIT 1` - just make the `limit` word upper

Reply:
oh got it

Reply:
fixed